### ISSUE TITLE
Problems with active Contao Debug mode

### ISSUE THREAD

**Issue by arno1979** [1up-lab/contao-efg#2]
efg is generating its own efg/config/config.php file when the a new form is created or the properties of a form are changed based on a template file.
This leads to an error if the Debug mode is activated in the Contao backend because debug info from `system/modules/efg/templates/internal/efg_internal_config.html5` is added to config.php.
It looks something like this (take notice of the first and the last lines):

```php
<!-- TEMPLATE START: system/modules/efg/templates/internal/efg_internal_config.html5 -->
<?php

/**
 * Contao Open Source CMS
 *
 * Copyright (c) 2005-2014 Leo Feyer
 *
 * @package   Efg
 * @author    Thomas Kuhn <mail@th-kuhn.de>
 * @license   http://www.gnu.org/licenses/lgpl-3.0.html LGPL
 * @copyright Thomas Kuhn 2007-2014
 */

// This file is created when saving a form in form generator
// last created on 2017-01-10 14:38:29


/**
* to fix height of style class w50 in backend
*/
if (TL_MODE == 'BE')
{
	$GLOBALS['TL_CSS'][] = 'system/modules/efg/assets/w50_fix.css';
}


/**
 * Use class ExtendedForm
 */
$GLOBALS['FE_MOD']['application']['form'] = 'ExtendedForm';
$GLOBALS['TL_CTE']['includes']['form'] = 'ExtendedForm';


/**
 * -------------------------------------------------------------------------
 * BACK END MODULES
 * -------------------------------------------------------------------------
 */



/**
 * -------------------------------------------------------------------------
 * FRONT END MODULES
 * -------------------------------------------------------------------------
 */

array_insert($GLOBALS['FE_MOD']['application'], count($GLOBALS['FE_MOD']['application']), array
(
	'formdatalisting' => 'ModuleFormdataListing'
));


/**
 * -------------------------------------------------------------------------
 * HOOKS
 * -------------------------------------------------------------------------
 */

$GLOBALS['TL_HOOKS']['processFormData'][] = array('FormdataProcessor', 'processSubmittedData');
$GLOBALS['TL_HOOKS']['outputFrontendTemplate'][] = array('FormdataProcessor', 'processConfirmationContent');
$GLOBALS['TL_HOOKS']['listComments'][] = array('FormdataComments', 'listComments');
$GLOBALS['TL_HOOKS']['getSearchablePages'][] = array('Formdata', 'getSearchablePages');
$GLOBALS['TL_HOOKS']['executePostActions'][] = array('Formdata', 'executePostActions');

<!-- TEMPLATE END: system/modules/efg/templates/internal/efg_internal_config.html5 -->
```
This could maybe fixed by temporarily disabling the debug mode similarly to this solution:
https://github.com/contao/core/commit/096a50eed4b4904d7ee189fa34555889cf3726ec

For instance:
Simply changing the first few lines of system/modules/efg/templates/internal/efg_internal_config.html5 to
```php
<?php
\Config::set('debugMode', false);
echo '<?php';
?>
```
fixes the problem.

**Comment by bytehead** [comment]
Thanks for submit.
Can you provide a pull request?

**Comment by arno1979** [comment]
I created the pull request. Sorry if it looks a bit messy with the three commits but this is my first PR... bear with me. :)

**Comment by bytehead** [comment]
Fixed in #3.

**Comment by bytehead** [comment]
I've released a new version: [3.0.2](https://github.com/1up-lab/contao-efg/releases/tag/3.0.2)
